app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java

```diff
@@ -726,8 +733,8 @@ public void getLocationFromLocationService() {
     public void onRequestPermissionsResult(int requestCode, String permissions[], int[] grantResults) {
         switch (requestCode) {
             case 1: {
-                if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED ) {
-                    LocationHelper locationHelper = new LocationHelper(MainActivity.this);
+                if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    locationHelper.getLocation();
```

Comment:
The location is stored in Preferences [here](https://github.com/fossasia/susi_android/blob/development/app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java#L736). When I implemented it, I stored the location in preference for a reason. I am not able to remember what exactly was the reason. Should probably change it.